src/include/uapi/ipc/topology.h

```diff
@@ -64,6 +64,7 @@ enum sof_comp_type {
 	SOF_COMP_EQ_FIR,
 	SOF_COMP_FILEREAD,	/**< host test based file IO */
 	SOF_COMP_FILEWRITE,	/**< host test based file IO */
+	SOF_COMP_SELECTOR,
```

Comment:
The driver will just see this a MUX type (since we are switching channel source and sinks). I've added a new "flavor" to ipc_comp_confg that allows FW to sub categorise the common components like MUX, mixer, volume etc into use case specific versions of the common components to save space, PM etc. @ranj063 is taken over this work atm

Reply:
I plan to make this change once @ranj063 PR#1004 is accepted & merged. (part of PR#993 rebase)

Reply:
ok, I will merge after #1004 is resolved.